db/dump.py

```diff
@@ -176,64 +171,81 @@ def generate_where(row_name, start_t=start_time, end_t=end_time):
-        with open(os.path.join(location, "model"), "w") as f:
-            logging.info(" - Copying table model...")
-            query = "SELECT %s FROM model %s" \
-                    % (", ".join(_TABLES["model"]), generate_where("date"))
-            cursor.copy_to(f, "(%s)" % query)
-
-
-        # highlevel
-        with open(os.path.join(location, "highlevel"), "w") as f:
-            logging.info(" - Copying table highlevel...")
-            cursor.copy_to(f, "(SELECT %s FROM highlevel %s)" %
-                           (", ".join(_TABLES["highlevel"]), generate_where("submitted")))
-
-        # highlevel_meta
-        with open(os.path.join(location, "highlevel_meta"), "w") as f:
-            logging.info(" - Copying table highlevel_meta...")
-            query = "SELECT %s FROM highlevel_meta WHERE id IN (SELECT id FROM highlevel %s)" \
-                    % (", ".join(_TABLES["highlevel_meta"]), generate_where("submitted"))
-            cursor.copy_to(f, "(%s)" % query)
-
-        # highlevel_model
-        with open(os.path.join(location, "highlevel_model"), "w") as f:
-            logging.info(" - Copying table highlevel_model...")
-            query = "SELECT %s FROM highlevel_model WHERE id IN (SELECT id FROM highlevel %s)" \
-                    % (", ".join(_TABLES["highlevel_model"]), generate_where("submitted"))
-            cursor.copy_to(f, "(%s)" % query)
-
-        # statistics
-        with open(os.path.join(location, "statistics"), "w") as f:
-            logging.info(" - Copying table statistics...")
-            cursor.copy_to(f, "(SELECT %s FROM statistics %s)" %
-                           (", ".join(_TABLES["statistics"]), generate_where("collected")))
-
-        # incremental_dumps
-        with open(os.path.join(location, "incremental_dumps"), "w") as f:
-            logging.info(" - Copying table incremental_dumps...")
-            cursor.copy_to(f, "(SELECT %s FROM incremental_dumps %s)" %
-                           (", ".join(_TABLES["incremental_dumps"]), generate_where("created")))
+        if dataset_dump:
+            # dataset
+            with open(os.path.join(location, "dataset"), "w") as f:
+                logging.info(" - Copying table dataset...")
+                cursor.copy_to(f, 'dataset', columns=_DATASET_TABLES['dataset'])
+
+            # dataset_class
+            with open(os.path.join(location, "dataset_class"), "w") as f:
+                logging.info(" - Copying table dataset_class...")
+                cursor.copy_to(f, 'dataset_class', columns=_DATASET_TABLES['dataset_class'])
+
+            # dataset_class_member
+            with open(os.path.join(location, "dataset_class_member"), "w") as f:
+                logging.info(" - Copying table dataset_class_member...")
+                cursor.copy_to(f, 'dataset_class_member', columns=_DATASET_TABLES['dataset_class_member'])
+
+        else:
+            # version
+            with open(os.path.join(location, "version"), "w") as f:
```

Comment:
Could we move these copy parts to a function called `copy_tables` and then call that for each table? Like [this](https://github.com/metabrainz/listenbrainz-server/blob/master/listenbrainz/db/dump.py#L293)